tools/mo/openvino/tools/mo/utils/ir_reader/layer_to_class.py

```diff
@@ -317,6 +317,8 @@ def copy_graph_with_ops(graph: Graph) -> Graph:
     new_graph = Graph()
     new_graph.stage = 'back'
     new_graph.graph = graph.graph
+    new_graph.inputs_order = graph.inputs_order
+    new_graph.outputs_order = graph.outputs_order
```

Comment:
Please double-check.
Not sure what do these two new attributes perform in the logic below? How do you preserve inputs and outputs order during restoring a graph if these attributes are not used below?

Reply:
IR reader runs prepare_emit_ir() during model saving. It is the same method that is used as in normal MO pipeline. 
prepare_emit_ir() runs serialize_network() where inputs/output order alignment is performed using graph.inputs_order and graph.outputs_order attributes:
https://github.com/openvinotoolkit/openvino/blob/8abb949af98cea64b766f5d0c9621f131bf02487/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L498
https://github.com/openvinotoolkit/openvino/blob/8abb949af98cea64b766f5d0c9621f131bf02487/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L539

So it is enough to set these attributes for inputs/outputs alignment.

Reply:
@popovaan, @achetver, are these attributes saved by POT in order to do the correct serialization?

Reply:
It was checked. We don't need to use these attributes directly in graph restoration, just provide them and they will used in MO emitter - inputs order:
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L539
and outputs order:
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py#L498

Reply:
> @popovaan, @achetver, are these attributes saved by POT in order to do the correct serialization?

They stored in restored graph. So if POT will not delete them they will be used in serialization.

Reply:
@achetver, OK. Please let them know of do not delete them just in case.